Python/_warnings.c

```diff
@@ -1020,10 +1023,9 @@ warn_explicit_with_fix(PyThreadState *tstate, PyObject *category, PyObject *mess
 
     if (rc == 1)  /* Already warned for this module. */
         goto return_none;
-    if (rc == 0) {
-        if (call_show_warning_with_fix(tstate, category, text, message, fix_txt, fix, filename,
-                              lineno, lineno_obj, sourceline, source) < 0)
-            goto cleanup;
+    if (rc == 0 || (call_show_warning_with_fix(tstate, category, text, message, fix_txt, fix, filename,
+                              lineno, lineno_obj, sourceline, source) < 0)) {
+        goto cleanup;
     }
     else /* if (rc == -1) */
```

Comment:
I think this `else` clause can now be deleted?

Reply:
Addressed in the second commit